components/tikv_util/src/time.rs

```diff
@@ -274,13 +285,30 @@ impl Instant {
             }
             Instant::MonotonicCoarse(t) => {
                 let now = monotonic_coarse_now();
-                Instant::elapsed_duration_coarse(now, t)
+                Instant::saturating_elapsed_duration_coarse(now, t)
+            }
+        }
+    }
+
+    pub fn saturating_elapsed(&self) -> Duration {
+        match *self {
+            Instant::Monotonic(t) => {
+                let now = monotonic_now();
+                Instant::saturating_elapsed_duration(now, t)
+            }
+            Instant::MonotonicCoarse(t) => {
+                let now = monotonic_coarse_now();
+                Instant::saturating_elapsed_duration_coarse(now, t)
             }
         }
     }
 
     pub fn elapsed_secs(&self) -> f64 {
```

Comment:
remove non-saturating functions?

Reply:
or just handle the jump back silently instead of adding a new function.

Reply:
I think I'll mark them as `[test]` to avoid misuse, but keep the interface here.